nexus/inventory/src/collector.rs

```diff
@@ -74,7 +77,7 @@ impl<'a> Collector<'a> {
 
     /// Collect inventory from all MGS instances
     async fn collect_all_mgs(&mut self) {
-        let clients = self.mgs_clients.clone();
+        let clients = self.mgs_clients.take().unwrap();
```

Comment:
This seems a _little_ fishy. What if `collect_one_mgs` didn't take self at all, and we passed in the logger and `in_progress` (the only things it uses AFAICT)? Then I think this method could be something like:

```rust
    async fn collect_all_mgs(&mut self) {
        for client in &self.mgs_clients {
            Self::collect_one_mgs(client, &self.log, &mut self.in_progress).await;
        }
    }
```

which should be fine borrow-check-wise.

Reply:
Ok, this is a *much* better solution. I think I'm just tired and grumpy today :). Changed in [cb59c68](https://github.com/oxidecomputer/omicron/pull/6810/commits/cb59c682d2b5843a45a244256b71654fdaa77249)

Thanks for the help @jgallagher!